pwiz_tools/Skyline/Model/Lib/BlibData/BlibDb.cs

```diff
@@ -267,11 +419,11 @@ public void Write()
                 int i = 0;
                 var sourceFiles = new Dictionary<string, long>();
                 var proteinTablesBuilder = new ProteinTablesBuilder(session);
-
+                using var spectrumInserter = new SpectrumInserter(session);
                 foreach (var spectrum in listSpectra)
                 {
                     var dbRefSpectrum = RefSpectrumFromPeaks(session, spectrum, sourceFiles);
-                    session.Save(dbRefSpectrum);
+                    spectrumInserter.InsertSpectrum(dbRefSpectrum);

```

Comment:
This SpectrumInserter change more than doubles the performance of this function, but it does disconnect the code from the NHibernate schema. Thus it has to be manually maintained if the schema is changed. Worth it? @brendanx67 @nickshulman

Reply:
Yes, it's probably worth it to write custom insert statements (although you might want to use more reflection in the implementation of that inserting code).
Sometimes, when you are worried about two pieces of code that might go out of sync with each other it's a good idea to write a unit test which makes sure they are the same.